src/main/msp/msp_box.c

```diff
@@ -377,8 +377,8 @@ int packFlightModeFlags(boxBitmask_t *mspFlightModeFlags)
     for (boxId_e boxId = 0; boxId < CHECKBOX_ITEM_COUNT; boxId++) {
         if (activeBoxIdGet(boxId)) {
             if (getBoxIdState(boxId))
-                bitArraySet(mspFlightModeFlags, mspBoxIdx);       // box is enabled
-            mspBoxIdx++;                                          // box is active, count it
+                bitArraySet(mspFlightModeFlags, boxId);       // box is enabled
+            mspBoxIdx++;                                      // box is active, count it
```

Comment:
Should we rename mspBoxIdx to mspActiveBoxCount ?

Reply:
Why? This is local variable, index into msp bitmask

Reply:
If we bitArraySet mspBoxIdx it will not transfer the bitmask, for a example read my answer down.
In any other code bitArraySet/Get and so on, we use the "real" bitmask or bitmask index.